core/win32/module_shared.c

```diff
@@ -1334,6 +1334,10 @@ find_remote_dll_base(HANDLE phandle, bool find64bit, char *dll_name)
     NTSTATUS res;
     uint64 addr = 0;
     char name[MAXIMUM_PATH];
+    /* FIXME - for 32-bit targets on a 64 bit machine, this loop doesn't return,
+     * if the dll is not loaded.
+     * When addr passes 0x800000000000 remote_query_virtual_memory_maybe64
+     * returns the previous mbi (ending at 0x7FFFFFFF0000) */
```

Comment:
Maybe it is best to put in the check you suggested to abort the loop (w/ this comment there as info or w/ XXX instead of FIXME): that seems the safest.

Reply:
Done